### DISCUSSION TITLE
Nuxt3 with redis client

### DISCUSSION THREAD

**Discussion by duderion** [nuxt/nuxt#16768] [Questions]
Hello !

we where trying to do the following:

in the file default.vue we have several $fetch commands that pull json from a server.

My idea was then to replace this fetch with my own "rfetch", that caches the API answer on a local redis server that is running beside the nuxt page.

Then i tried node-redis and ioredis and both didnt work (Undefined Buffer, ChartAt process.version undef...). Maybe because our .vue file is running in the browser, not on the node backend.

My question is now: Where shall i put my function to make sure that the redis communication works, so "server side", And that it’s available inside .vue files ?

another idea might be, to extend ohmyfetch inside the response and request hooks to read and write from/to redis.  But I don’t know if I can access redis db here .

Regards,

Adrian

**Comment by sevillaarvin** [comment]
What does your implementation look like? Maybe put the redis logic only in `~/server` folder so that it will only be executed server side.

**Comment by duderion** [comment]
We got the tip from Seabstian Chopin himself that we can reach local servers through the nuxt api ts files. So we put our cache logic there and changed the routes from server/api to local/api. Then the local api does the cache logic. It’s 🙌 super fast
